test/Recollateralization.test.ts

```diff
@@ -3527,14 +3573,73 @@ describe(`Recollateralization - P${IMPLEMENTATION}`, () => {
         // Advance time till auction ended
         await advanceTime(config.auctionLength.add(100).toString())
 
-        // Close final auction - Haircut will be taken
+        // Close 2nd to final auction
         await expectEvents(facadeTest.runAuctionsForAllTraders(rToken.address), [
           {
             contract: backingManager,
             name: 'TradeSettled',
             args: [anyValue, token1.address, backupToken1.address, sellAmt4, minBuyAmt4],
             emitted: true,
           },
+          {
+            contract: backingManager,
+            name: 'TradeStarted',
+            args: [anyValue, token1.address, backupToken1.address, anyValue, anyValue],
+            emitted: true,
+          },
+        ])
+
+        auctionTimestamp = await getLatestBlockTimestamp()
+
+        // Check new auction
+        // Token1 -> Backup Token 1 Auction
```

Comment:
can we guarantee that we will be able to finalize trading? what is causing 2 of the same auction to run back-to-back (token 1 => backup 1), and how do we know we won't keep running these ad infinitum?

Reply:
Wrote up an explanation in comments...